src/middleware/normalize.rs

```diff
@@ -0,0 +1,93 @@
+//! `Middleware` to normalize request's URI
+
+use regex::Regex;
+use actix_service::{Service, Transform};
+use futures::future::{self, FutureResult};
+
+use crate::service::{ServiceRequest, ServiceResponse};
+
+lazy_static::lazy_static! {
+    static ref MERGE_SLASH: Regex = Regex::new("//+").unwrap();
+}
+
+#[derive(Default, Clone, Copy)]
+/// `Middleware` to normalize request's URI in place
+///
+/// Performs following:
+///
+/// - Merges double slashes into one.
+/// - Removes trailing slash.
+pub struct InPlace;
```

Comment:
i dont like name, let's name it `Normalize` or something like this

Reply:
Ok, but I'll name it `NormalizePath`